changelog_unreleased/javascript/13677.md

```diff
@@ -0,0 +1,32 @@
+#### Fix printing dangling comments for `break|continue|debugger` (#13677 by @liuxingbaoyu)
+
+<!-- Optional description if it makes sense. -->
+
+<!-- prettier-ignore -->
+```js
+// Input
+
+for (;;) {
+  if (a) break; // comment
+  else foo();
+  if(a) continue; // comment
+  else foo();
+  if(a) debugger; // comment
+  else foo();
+}
+
+// Prettier stable
+
+Error
+
+// Prettier main
+
+for (;;) {
+  if (a) break; // comment
+  else foo();
+  if(a) continue; // comment
+  else foo();
+  if(a) debugger; // comment
+  else foo();
+}
+```
```

Comment:
````suggestion
<!-- prettier-ignore -->
```js
// Input
for (;;) {
  if (a) break; // comment
  else foo();
  if(a) continue; // comment
  else foo();
  if(a) debugger; // comment
  else foo();
}

// Prettier stable
Error

// Prettier main
for (;;) {
  if (a) break; // comment
  else foo();
  if(a) continue; // comment
  else foo();
  if(a) debugger; // comment
  else foo();
}
```
````